actions/drain_node_handler.go

```diff
@@ -119,36 +130,40 @@ func (h *drainNodeHandler) evictNodePods(ctx context.Context, log logrus.FieldLo
 
 	log.Infof("evicting %d pods", len(pods))
 
-	if err := h.sendPodsRequests(ctx, pods, h.evictPod); err != nil {
+	if err := h.sendPodsRequests(ctx, metav1.DeleteOptions{}, pods, h.evictPod); err != nil {
 		return fmt.Errorf("sending evict pods requests: %w", err)
 	}
 
 	return h.waitNodePodsTerminated(ctx, node)
 }
 
-func (h *drainNodeHandler) deleteNodePods(ctx context.Context, log logrus.FieldLogger, node *v1.Node) error {
+func (h *drainNodeHandler) deleteNodePods(ctx context.Context, log logrus.FieldLogger, options metav1.DeleteOptions, node *v1.Node) error {
```

Comment:
I wonder have you considered using functional options pattern? https://golang.cafe/blog/golang-functional-options-pattern.html